qemu/tests/qemu_guest_agent.py

```diff
@@ -2686,6 +2686,63 @@ def bl_check(qga_cmd):
             cmd = "mv -f /etc/sysconfig/qemu-ga-bk /etc/sysconfig/qemu-ga"
             session.cmd(cmd)
 
+    @error_context.context_aware
+    def gagent_check_virtio_device(self, test, params, env):
+        """
+        Get hostname, timezone and currently active users on the vm.
+
+        :param test: kvm test object
+        :param params: Dictionary with the test parameters
+        :param env: Dictionary with test environment.
+        """
+        session = self._get_session(params, None)
+        self._open_session_list.append(session)
+
+        def _result_check(rsult_qga, rsult_guest):
+            if rsult_qga != rsult_guest:
+                msg = "The result is different between qga and guest\n"
+                msg += "from qga: %s\n" % rsult_qga
+                msg += "from guest: %s\n" % rsult_guest
+                test.fail(msg)
+
+        devs_list = self.gagent.get_virtio_device()
+        for device in devs_list:
+            driver_name = device["driver-name"]
+            error_context.context("Check %s info." % driver_name, logging.info)
+
+            driver_date = device["driver-date"]
+            driver_version = device["driver-version"]
+            device_address = device["address"]["data"]
+            device_id = device_address["device-id"]
+            vendor_id = device_address["vendor-id"]
+            check_driver_cmd = params["check_driver_powershell_cmd"]
+            if "Ethernet" in driver_name:
+                check_driver_cmd = check_driver_cmd % ("friendlyname",
+                                                       driver_name)
+            else:
+                check_driver_cmd = check_driver_cmd % ("devicename",
+                                                       driver_name)
+            driver_info_guest = session.cmd_output(check_driver_cmd)
+            # check driver date
+            date = re.search(r"driverdate.*\:\s(\S{8})", driver_info_guest,
+                             re.I).group(1)
+            date_l = [date[0:4], date[4:6], date[6:8]]
+            driver_date_guest = "-".join(date_l)
+            _result_check(driver_date, driver_date_guest)
+
+            # check driver version
+            driver_ver_guest = re.search(r"driverversion.*\:\s(\S+)",
```

Comment:
if the version is a digit, better to use `\d` instead of `\S`

Reply:
driverversion : 100.82.104.18000